### PR TITLE
[skip ci] Menu entry for REST API

### PR DESCRIPTION
**Pull request by Nyco**
We had: 
  - 'REST API for client developers': 'REST-API.md'
  - 'Operation and maintenance':
      - 'HTTP Administration API': 'http-api/http-administration-api-documentation.md'
  - "Developer's guide":
      - 'REST interface to metrics': 'developers-guide/REST-interface-to-metrics.md'

Now we have a menu entry for "REST API", for Client, Admin, and Metrics.

### CODE REVIEW

**Comment by fenek** [comment]
Looks good! Can you please also:

1. Create a new directory like `rest-api` and move these docs there
2. Rename files to match new titles
3. Update links in other files with new filenames (e.g. in README.md)

**Comment by Nyco** [comment]
I think it is good. Can we test the branch?

File: doc/advanced-configuration/Listener-modules.md [hunk 1]
```diff
@@ -82,7 +82,7 @@ Unlike `ejabberd_c2s`, it doesn't use `ejabberd_receiver` or `ejabberd_listener`
             `{"_", "/ws-xmpp", mod_websockets, []}`
 
     * <i>(OBSOLETE)</i> `mongoose_api` - REST API for accessing internal MongooseIM metrics.
-        Please refer to [REST interface to metrics](../developers-guide/REST-interface-to-metrics.md)
+        Please refer to [REST interface to metrics](../rest-api/Metrics-backend.md)
```
Comment (goddammit) [inline doc/advanced-configuration/Listener-modules.md]: Please refer to the [REST interface to metrics] article ?

File: doc/rest-api/Client-frontend.md [hunk 2]
```diff
@@ -51,14 +53,17 @@ REST API functionalities are enabled and exposed on the given paths.
 By default the REST API is exposed on port 8089 but this can be changed to whatever is more convenient.
 
 For more details about possible `ejabberd_cowboy` configuration parameters please
-see the relevant documentation in [Listener modules](advanced-configuration/Listener-modules.md#http-based-services-bosh-websocket-rest-ejabberd_cowboy)
+see the relevant documentation in [Listener modules](../advanced-configuration/Listener-modules.md#http-based-services-bosh-websocket-rest-ejabberd_cowboy).
```
Comment (goddammit) [inline doc/rest-api/Client-frontend.md]: see the relevant documentation in the [Listener modules]?

**Comment by deadjdona** [comment]
iframe is not working for swagger, looks like pasted in visual editor.

**Pull request merged**